File: javascript/bob/bob.js
export const hey = (message) => {
  message = message.trim();

  const isQuestion = message.endsWith("?");
  const isYelling = message === message.toUpperCase() && message !== message.toLowerCase();
  const isSilence = message.length === 0;

  if(isSilence)                     return "Fine. Be that way!";
  else if(isQuestion && isYelling)  return "Calm down, I know what I'm doing!"; 
  else if(isQuestion)               return "Sure.";
  else if(isYelling)                return "Whoa, chill out!";
  else                              return "Whatever.";
};

File: python/bob/bob.py
def response(hey_bob: str) -> str:
    is_yelling = hey_bob.isupper()
    is_asking_a_question = hey_bob.strip().endswith("?")
    
    if is_yelling and is_asking_a_question:
        return "Calm down, I know what I'm doing!"
    
    if is_yelling:
        return "Whoa, chill out!"
    
    if is_asking_a_question:
        return "Sure."
    
    if hey_bob.isspace() or not hey_bob:
        return "Fine. Be that way!"
    
    return "Whatever."
